daskms/experimental/zarr/tests/test_zarr.py

```diff
@@ -429,6 +429,38 @@ def test_xarray_reading_daskms_written_dataset(ms, tmp_path_factory):
     path = store / "test.zarr"
     dask.compute(xds_to_zarr(datasets, path, consolidated=True))
 
+    extra = tmp_path_factory.mktemp("extra")
+    datasets[0].to_zarr(extra)
+
     for i, mem_ds in enumerate(datasets):
         ds = xarray.open_zarr(path / "MAIN" / f"MAIN_{i}")
-        assert ds == mem_ds
+
+        for k in (set(ds.data_vars) | set(mem_ds.data_vars)) - {"ROWID"}:
+            assert_array_equal(ds.data_vars[k], mem_ds.data_vars[k])
+
+        for k in (set(ds.coords) | set(mem_ds.coords)) - {"ROWID"}:
+            assert_array_equal(ds.coords[k], mem_ds.coords[k])
+
+        # capitalised ROWID breaks the lowercase
+        # xarray coordinate naming convention
+        # so xarray treats it as a data variable
+        assert_array_equal(ds.data_vars["ROWID"], mem_ds.coords["ROWID"])
+
+        attr_keys = (set(mem_ds.attrs) | set(ds.attrs)) - {
+            DASKMS_PARTITION_KEY,
+            DASKMS_ATTR_KEY,
+        }
+
+        for k in attr_keys:
+            assert ds.attrs[k] == mem_ds.attrs[k]
+
+        # DASKMS_ATTR_KEY is added by dask-ms when writing
+        # so xarray reads it in
+        assert DASKMS_ATTR_KEY in ds.attrs
+        assert DASKMS_ATTR_KEY not in mem_ds.attrs
+
+        # xarray converts tuples to list when json encoding
+        assert (
+            tuple(map(tuple, ds.attrs[DASKMS_PARTITION_KEY]))
+            == mem_ds.attrs[DASKMS_PARTITION_KEY]
+        )
```

Comment:
I'm happy that this test case test equality on variables, coordinates and most attrs. The test case handles some json encoding discrepancies for the `DASMS_PARTITION_KEY` and tests for its existence on the `xarray.open_dataset` version.